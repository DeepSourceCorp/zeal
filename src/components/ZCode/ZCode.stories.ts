import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'
import ZCode from './ZCode.vue'

export default {
  title: 'Code',
  component: ZCode,
  excludeStories: /.*Data$/
}

export const DefaultCode = () => ({
  components: { ZCode },
  data() {
    return {
      content: `<div class='input-container'>
        <z-code :content="content"></z-code>           
    </div>`
    }
  },
  template: `<div class='input-container'>
        <z-code :content="content"></z-code>           
    </div>`
})
