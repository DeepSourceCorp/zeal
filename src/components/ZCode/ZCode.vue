<template>
  <div class="z-code" v-html="content"></div>
</template>
<script>
export default {
  name: 'ZCode',
  props: {
    content: {
      type: String
    }
  }
}
</script>
<style>
@import '../../assets/css/syntax_highlighter.css';
</style>
