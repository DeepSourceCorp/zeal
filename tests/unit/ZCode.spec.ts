import ZCode from '../../src/components/ZCode'
import { mount } from '@vue/test-utils'

describe('ZCode', () => {
  it('renders a code component with Default Syntax highlighting', async () => {
    const wrapper = mount(ZCode, {
      propsData: {
        content: `<div class="highlight"><pre><span class="ln">2</span>\n<span class="ln">3</span>\n<span class="ln">4</span><span class="nd">@dramatiq.actor</span>\n<span class="hl"><span class="ln">5</span><span class="k">def</span> <span class="nf">example</span><span class="p">(</span><span class="p">)</span><span class="p">:</span></span><span class="ln">6</span>    <span class="k">pass</span>\n</pre></div>`
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.classes()).toContain('z-code')
  })
})
