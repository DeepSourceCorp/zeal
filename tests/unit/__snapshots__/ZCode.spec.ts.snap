// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZCode renders a code component with Default Syntax highlighting 1`] = `
<div class="z-code">
  <div class="highlight"><pre><span class="ln">2</span>
<span class="ln">3</span>
<span class="ln">4</span><span class="nd">@dramatiq.actor</span>
<span class="hl"><span class="ln">5</span><span class="k">def</span> <span class="nf">example</span><span class="p">(</span><span class="p">)</span><span class="p">:</span></span><span class="ln">6</span>    <span class="k">pass</span>
</pre>
  </div>
</div>
`;
